src/Service/Database/Shell.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+namespace Meanbee\Magedbm2\Service\Database;
+
+use Meanbee\Magedbm2\Application;
+use Meanbee\Magedbm2\Helper\TablePatternExpander;
+use Meanbee\Magedbm2\Exception\ConfigurationException;
+use Meanbee\Magedbm2\Service\DatabaseInterface;
+use Meanbee\Magedbm2\Exception\ServiceException;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\Process;
+
+class Shell implements DatabaseInterface
+{
+    /**
+     * This is the amount of time that a process will be allowed to execute for.
+     */
+    const PROCESS_TIMEOUT_SECONDS = 3600;
+
+    /**
+     * @var Application
+     */
+    private $app;
+
+    /**
+     * @var Application\ConfigInterface
+     */
+    private $config;
+
+    /**
+     * @var OutputInterface
+     */
+    private $outputInterface;
+
+    /**
+     * @var TablePatternExpander
+     */
+    private $tablePatternExpander;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(Application $app, Application\ConfigInterface $config = null)
+    {
+        $this->app = $app;
+        $this->config = $config;
+        $this->tablePatternExpander = new TablePatternExpander();
+        $this->logger = new NullLogger();
+    }
+
+    /**
+     * @param OutputInterface $interface
+     */
+    public function setOutputAdapter(OutputInterface $interface)
+    {
+        $this->outputInterface = $interface;
+    }
+
+
+    /**
+     * Import the given backup file into the database.
+     *
+     * @param string $file Path to a file to import.
+     *
+     * @return void
+     * @throws \Symfony\Component\Process\Exception\RuntimeException
+     * @throws ServiceException
+     */
+    public function import($file)
+    {
+        $command = array_merge(
+            ['zcat ' . $file . ' | mysql'],
```

Comment:
`zcat` doesn't work for `.gz` files on MacOS, as it expects the input file to have a `.z` extension. For `.gz` files on MacOS, you need to `gzcat`, but that's not available in Linux distros. Most platform compatible option in this case is to just use `gunzip -c`.